### PR TITLE
DIRECTOR: Fixes for D5 + sprite rendering

### PR DESCRIPTION
**Pull request by moralrecordings**

### CODE REVIEW

File: engines/director/score.cpp [hunk 1]
```diff
@@ -551,6 +551,7 @@ void Score::update() {
 }
 
 void Score::renderFrame(uint16 frameId, RenderMode mode) {
+	uint32 start = g_system->getMillis(false);
```
Comment (sev-) [inline engines/director/score.cpp:554]: Why are you skipping recording this value?

File: engines/director/castmember/bitmap.cpp [hunk 2]
```diff
@@ -682,6 +675,15 @@ void BitmapCastMember::setPicture(Image::ImageDecoder &image, bool adjustSize) {
 	setModified(true);
 }
 
+Common::Point BitmapCastMember::getRegistrationOffset() {
+	return Common::Point(_regX - _initialRect.left, _regY - _initialRect.top);
+}
+
+Common::Point BitmapCastMember::getRegistrationOffset(int16 width, int16 height) {
```
Comment (sev-) [inline engines/director/castmember/bitmap.cpp:682]: I think the better names would be `currentWidth` and `currentHeight`, e.g. as I understand, these are the current dimensions, when the sprite was stretched.

File: engines/director/castmember/castmember.h [hunk 3]
```diff
@@ -86,6 +86,11 @@ class CastMember : public Object<CastMember> {
 
 	virtual Common::String formatInfo() { return Common::String(); };
 
+	virtual Common::Rect getBbox();
+	virtual Common::Rect getBbox(int16 width, int16 height);
```
Comment (sev-) [inline engines/director/castmember/castmember.h:90]: Same logic here, please describe it in a doxygen comment

File: engines/director/castmember/filmloop.cpp [hunk 4]
```diff
@@ -88,7 +89,7 @@ Common::Array<Channel> *FilmLoopCastMember::getSubChannels(Common::Rect &bbox, C
 		int16 width = src._width * widgetRect.width() / _initialRect.width();
 		int16 height = src._height * widgetRect.height() / _initialRect.height();
 
-		Channel chan(&src);
+		Channel chan(nullptr, &src);
```
Comment (sev-) [inline engines/director/castmember/filmloop.cpp:92]: please add here a comment explaining why Score is `nullptr` here.

**Review by sev- — CHANGES_REQUESTED** [review]
Have some notes

File: engines/director/score.cpp (hunk 1 above)
Comment (moralrecordings) [reply engines/director/score.cpp:554]: The time is only used for the debug log, isn't the idea to only record the values which affect the control flow or playback?

File: engines/director/castmember/bitmap.cpp (hunk 2 above)
Comment (moralrecordings) [reply engines/director/castmember/bitmap.cpp:682]: Fixed

File: engines/director/castmember/filmloop.cpp (hunk 4 above)
Comment (moralrecordings) [reply engines/director/castmember/filmloop.cpp:95]: Done

File: engines/director/castmember/castmember.h (hunk 3 above)
Comment (moralrecordings) [reply engines/director/castmember/castmember.h]: Done

**Comment by sev-** [comment]
Thank you!

**Pull request merged**